feat: Add loyalty cards feature

Edigorin opened the pull request:
## Summary

Add support for loyalty cards/membership cards that users can store and quickly access from the shopping list page via a wallet FAB.

Closes #462

## Changes

### Backend
- Add `LoyaltyCard` model with name, barcode data/type, description, color
- Add CRUD API endpoints for loyalty cards (`/api/household/{id}/loyalty-card`)
- Add database migration for `loyalty_cards` table
- Add household feature flag for loyalty cards
- Add loyalty card import functionality
- Add comprehensive API tests

### Frontend
- Add loyalty cards list page
- Add loyalty card detail page with fullscreen barcode display
- Add loyalty card add/edit page with barcode scanning
- Add barcode scanning from camera (mobile) and webcam (web)
- Add barcode scanning from gallery/image upload
- Add wallet FAB on shopping list page to access loyalty cards
- Support multiple barcode types: CODE128, QR, EAN13, EAN8, UPC-A, UPC-E, CODE39, ITF, Codabar, DataMatrix, Aztec, PDF417
- Add localization strings for loyalty cards feature
- Add loyalty cards option to import settings

## Testing

- Added `backend/tests/api/test_api_loyalty_card.py` with comprehensive API tests
- Manually tested on Android, and Web platforms. No ability to test on IOS

<img width="2560" height="1389" alt="chrome_99653v6vYu" src="https://github.com/user-attachments/assets/b75b9077-abe0-4efb-8ece-0a99a312f128" />
<img width="2560" height="1389" alt="chrome_hfh8m63lnl" src="https://github.com/user-attachments/assets/c3cdc22f-1499-48df-a138-7694ed9cd170" />
<img width="2560" height="1389" alt="chrome_knKkUHGaEP" src="https://github.com/user-attachments/assets/ceceac58-66cc-487a-a632-c622bdc6666e" />
<img width="2560" height="1389" alt="chrome_nYkMRzRzgm" src="https://github.com/user-attachments/assets/bceb9bf8-f719-4c44-bbc7-1bc8115325af" />
<img width="2560" height="1389" alt="chrome_Zl9xBpXxYH" src="https://github.com/user-attachments/assets/8d54d9fe-72de-49a6-8a3c-114576ccf36a" />

Edigorin committed:
feat: Add loyalty cards feature

Add support for loyalty cards/membership cards that users can store
and access from the shopping list page via a wallet FAB.

Backend:
- Add LoyaltyCard model with name, barcode data/type, description, color
- Add CRUD API endpoints for loyalty cards
- Add database migration for loyalty_cards table
- Add household feature flag for loyalty cards
- Add loyalty card import functionality
- Add comprehensive API tests

Frontend:
- Add loyalty cards list page with search functionality
- Add loyalty card detail page with fullscreen barcode display
- Add loyalty card add/edit page with barcode scanning
- Add barcode scanning from camera (mobile) and webcam (web)
- Add barcode scanning from gallery/image upload
- Add wallet FAB on shopping list page to access loyalty cards
- Support multiple barcode types: CODE128, QR, EAN13, etc.
- Add localization strings for loyalty cards feature
- Add loyalty cards import settings option

Closes #462

Edigorin committed:
fix: Remove unnecessary import and fix household serialization

- Remove redundant dart:typed_data import (Uint8List already provided by flutter/foundation.dart)
- Fix featureLoyaltyCards deserialization to preserve null values (fixes failing test)

josephclaytonhansen commented:
Love this, hope to see it in!

TomBursch reviewed (commented):
Hey, thanks for the PR. Because this is such a huge change, I didn't have time until now to review it. The code looks really good, and I'm happy to merge it.

I just have some minor questions:

Why does the loyalty model store a photo but it's never set to anything and not used? I also can't upload a loyalty card without a barcode.

And why do you implement the barcode scanning on the web yourself? Why not use the web functionality of the mobile scanner? https://pub.dev/packages/mobile_scanner#web

Edigorin committed:
fix: Address PR review — remove unused photo field, make barcode optional, remove custom web scanner

- Remove photo/photo_file column from loyalty card model, migration, schemas, and controller
- Remove photo-related relationships from File model
- Make barcode_type and barcode_data nullable in model, migration, and schemas
- Allow creating loyalty cards without a barcode (frontend validation removed)
- Conditionally render barcode UI only when barcode data is present
- Delete custom web barcode scanner (web_barcode_scanner.dart, barcode_scanner.js)
- Remove html5-qrcode CDN script from index.html
- Hide gallery scan button on web (analyzeImage not supported)
- Use mobile_scanner's built-in web support for camera scanning
- Add test for creating loyalty card without barcode

Edigorin committed:
fix(l10n): deduplicate ARB keys and add missing localization entries

Remove 57-line duplicate block in app_en.arb, add 14 new loyalty card
value keys with metadata, and localize 3 hardcoded strings in
loyalty_card_page.dart and loyalty_card_list_page.dart.

Ultraworked with [Sisyphus](https://github.com/code-yeongyu/oh-my-opencode)
Co-authored-by: Sisyphus <clio-agent@sisyphuslabs.ai>

Edigorin committed:
fix: improve type safety and backward compat in shopping list FAB

Change featureLoyaltyCards default from true to false for backward
compatibility when backend omits the field, and replace dynamic with
Household type annotation.

Ultraworked with [Sisyphus](https://github.com/code-yeongyu/oh-my-opencode)
Co-authored-by: Sisyphus <clio-agent@sisyphuslabs.ai>

Edigorin committed:
fix(backend): use save() pattern and explicit imports in loyalty card

Replace db.session.add/commit with card.save() to match existing import
services, switch to absolute import for LoyaltyCard model, and replace
wildcard import with explicit named imports in controller init.

Ultraworked with [Sisyphus](https://github.com/code-yeongyu/oh-my-opencode)
Co-authored-by: Sisyphus <clio-agent@sisyphuslabs.ai>

thieneret commented:
I would really like this feature, so I would like to ask if there is a chance that this PR will be merged in the near future?
